widget/entry.go

```diff
@@ -525,6 +555,8 @@ func (e *Entry) TappedSecondary(pe *fyne.PointEvent) {
 	clipboard := fyne.CurrentApp().Driver().AllWindows()[0].Clipboard()
 	super := e.super()
 
+	undoItem := fyne.NewMenuItem("Undo", e.Undo)
+	redoItem := fyne.NewMenuItem("Redo", e.Redo)
```

Comment:
How common is it to have undo and redo in the context menu? It seems to me like most applications don't expose it there.

Reply:
QT has undo/redo in the context menu in every text box by default. Also all popular web browsers. GTK apps are the only counter example I can think of. I don't think it's unreasonable to expose it in the context menu.

Reply:
A possible improvement though would be to only show undo/redo if there is actually anything to undo/redo

Reply:
> A possible improvement though would be to only show undo/redo if there is actually anything to undo/redo

I just implemented that, it was pretty simple

Reply:
Normally it would be enable/disable rather than add/remove - having menus be consistent in size and layout makes it easier to navigate them at speed.

Reply:
I mean, it *is* less cluttered. Also that's how Chromium does it. And if you want to operate at speed, you'll want to use keyboard shortcuts anyway. But your project, your design decisions :)